src/Symfony/Component/Cache/Adapter/ChainAdapter.php

```diff
@@ -73,7 +73,9 @@ static function ($sourceItem, $item, $sourceMetadata = null) use ($defaultLifeti
                 $item->isHit = $sourceItem->isHit;
                 $item->metadata = $item->newMetadata = $sourceItem->metadata = $sourceMetadata;
 
-                if (0 < $defaultLifetime) {
+                if (isset($item->metadata[CacheItem::METADATA_EXPIRY])) {
+                    $item->expiresAfter((int) (0.1 + $item->metadata[CacheItem::METADATA_EXPIRY] - time()));
```

Comment:
what about `\DateTime::createFromFormat('U.u', $item->metadata[CacheItem::METADATA_EXPIRY])` ?

Reply:
Thank you. 

I thought about that. I decided not to (without measuring) because I didnt want to create a DateTime object that would just be created back to a int.

Reply:
I understand. I think I'd still suggest using the object because of the increased precision it allows (the component handles the sub-second part of the expiry, as supported by DateTime)

Reply:
Sure. Thank you for your input.